dd-sdk-android/src/test/kotlin/com/datadog/android/core/configuration/ConfigurationBuilderTest.kt

```diff
@@ -103,12 +105,17 @@ internal class ConfigurationBuilderTest {
                 plugins = emptyList()
             )
         )
-        assertThat(config.tracesConfig).isEqualTo(
-            Configuration.Feature.Tracing(
-                endpointUrl = DatadogEndpoint.TRACES_US,
-                plugins = emptyList()
+        assertThat(config.tracesConfig)
+            .isEqualToIgnoringGivenFields(
```

Comment:
why cannot we use `isEqual`? is this because normally `NoOpSpanEventMapper() != NoOpSpanEventMapper()`? Maybe for `NoOp` instances we could use `object` and not `class`? what do you think? they are stateless anyway.

Reply:
Is because when comparing those fields it will use `compare by reference` and not `by type`. Using `Object` instead of `class` will work but that will imply changing the `NoOp` generator...and I would rather to this in another PR as it is out of this topic.

Reply:
yes, for sure, such change shouldn't be a part of this PR